lib/internal/errors.js

```diff
@@ -64,6 +65,16 @@ const prepareStackTrace = (globalThis, error, trace) => {
     return f(error, trace);
   }
 
+  for (let l = trace.length - 1; l >= 0; l--) {
```

Comment:
This PR seems to move the "overhead" from success-path to exception-path. It might be a good idea to benchmark both paths, say, for `buf.write()` method, just to be able to understand whether there is any improvement/regression.

Reply:
misc/hidestackframes.js n=100000 type="hide-err": 22,456.052314327284
misc/hidestackframes.js n=100000 type="direct-err": 20,387.649071345215
misc/hidestackframes.js n=100000 type="hide-noerr": 10,342,015.627819816
misc/hidestackframes.js n=100000 type="direct-noerr": 10,862,952.985139482


```js
'use strict';

const common = require('../common.js');

const bench = common.createBenchmark(main, {
  type: ['hide-err', 'direct-err', 'hide-noerr', 'direct-noerr'],
  n: [100000]
}, {
  flags: ['--expose-internals']
});

function main({ n, type }) {
  const {
    hideStackFrames,
    codes: {
      ERR_INVALID_ARG_TYPE,
    },
  } = require('internal/errors');

  const testfn = (value) => {
    if (typeof value !== 'number') {
      throw new ERR_INVALID_ARG_TYPE('Benchmark', 'number', value);
    }
    let r = 1;
    for (let i = 0; i < 4; i++)
      r *= (i / 2);
    return r;
  };

  let fn = testfn;
  if (type.startsWith('hide'))
    fn = hideStackFrames(testfn);
  let value = 42;
  if (type.endsWith('-err'))
    value = 'err';

  bench.start();

  for (let i = 0; i < n; i++) {
    try {
      fn(value);
    } catch (e) {
      // No-op
    }
  }

  bench.end(n);
}
```

Reply:
The error path is already far too slow for the regression to be noticeable, there is a factor of 500

Reply:
Without the PR:
misc/hidestackframes.js n=100000 type="hide-err": 24,434.469081672654
misc/hidestackframes.js n=100000 type="direct-err": 24,511.501403118
misc/hidestackframes.js n=100000 type="hide-noerr": 6,504,481.750538165
misc/hidestackframes.js n=100000 type="direct-noerr": 10,724,409.173530914

Reply:
I have added a small arithmetic loop to your test so that it is more realistic, otherwise the difference is huge but unrealistic

Reply:
This is without the arithmetic operations:
```
mmom@mmom-workstation:~/src/node$ ./node benchmark/misc/hidestackframes.js  # without PR
misc/hidestackframes.js n=100000 type="hide-err": 24,542.416148820295
misc/hidestackframes.js n=100000 type="direct-err": 24,689.453077934282
misc/hidestackframes.js n=100000 type="hide-noerr": 7,136,211.80048265
misc/hidestackframes.js n=100000 type="direct-noerr": 10,672,188.884082846
mmom@mmom-workstation:~/src/node$ ./node benchmark/misc/hidestackframes.js    # with PR
misc/hidestackframes.js n=100000 type="hide-err": 22,586.849942378212
misc/hidestackframes.js n=100000 type="direct-err": 20,270.969751583732
misc/hidestackframes.js n=100000 type="hide-noerr": 11,406,198.333485987
misc/hidestackframes.js n=100000 type="direct-noerr": 11,542,427.250100967
mmom@mmom-workstation:~/src/node$ 
```

Reply:
It may be a good idea to place the benchmark into `benchmark/misc`. Then you could use compare script to compare new vs old performance. See:
https://github.com/nodejs/node/blob/master/doc/guides/writing-and-running-benchmarks.md#comparing-nodejs-versions

Reply:
Here's a benchmark run in CI. https://ci.nodejs.org/view/Node.js%20benchmark/job/benchmark-node-micro-benchmarks/668/